src/services/sections/section.service.js

```diff
@@ -204,8 +211,18 @@ class SectionService {
 			throw new Forbidden(this.err.noAccess);
 		}
 
-		const { _id } = await service.patch(id, data, internParams);
-		return setUserScopePermission({ _id }, 'write');
+		const { _id, state } = await service.patch(id, this.manageStateDiffsIfProvided(data), internParams);
+
+		const response = { _id };
+		const queryState = params.query.state;
+		if (data.stateDiff && ['all', 'diff'].includes(queryState)) {
```

Comment:
check if stateDiff is set could be done in if that checks the queryparamteres. So array include has to run only once